six/modules/c++/six.sicd/source/Utilities.cpp

```diff
@@ -507,6 +507,24 @@ Vector3 Utilities::getGroundPlaneNormal(const ComplexData& data)
 
     return groundPlaneNormal.unit();
 }
+
+std::auto_ptr<ComplexData> Utilities::readXML(
+        const std::string& xmlPathname)
+{
+    XMLControlFactory::getInstance().addCreator(
+        DataType::COMPLEX,
+        new XMLControlCreatorT<ComplexXMLControl>());
+
+    std::auto_ptr<ComplexData> complexData(reinterpret_cast<ComplexData*>(
+            six::readXML(xmlPathname, DataType::COMPLEX).release()));
+
+    if (complexData.get() == NULL)
```

Comment:
This will never happen (assuming `six::readXML()` doesn't have a bug where it returns a NULL `auto_ptr`).  `dynamic_cast` would give you a NULL pointer if it was the wrong type but not `reinterpret_cast`.  I think you can stick with `reinterpret_cast` and assume that `six::readXML()` did the right thing when you passed in `COMPLEX`.